test/integration/integration_test.go

```diff
@@ -0,0 +1,90 @@
+//go:build integration
+
+package main
+
+import (
+	"context"
+	"fmt"
+	"kubearchive/test"
+	"os"
+	"testing"
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func TestMain(m *testing.M) {
+	if os.Getenv("KO_DOCKER_REPO") == "" {
+		os.Setenv("KO_DOCKER_REPO", "kind.local")
+	}
+	os.Exit(m.Run())
+}
+
+func TestSomething(t *testing.T) {
+	namespaceName := fmt.Sprintf("test-%s", test.RandomString())
+	namespace := fmt.Sprintf(`
+---
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: %s
+`, namespaceName)
+
+	deployment := fmt.Sprintf(`
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: kubearchive
+  namespace: %s
+  labels:
+    app.kubernetes.io/name: kubearchive
+spec:
+  selector:
+    matchLabels:
+      app.kubernetes.io/name: kubearchive
+  template:
+    metadata:
+      labels:
+        app.kubernetes.io/name: kubearchive
+    spec:
+      containers:
+      - name: kubearchive
+        image: ko://kubearchive/cmd/api/
+        ports:
+        - containerPort: 8081
+`, namespaceName)
+
+	err := test.CreateResources(namespace, deployment)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	client, err := test.GetKubernetesClient()
+
+	acc := 0.0
+	for {
+		if acc >= 30 {
+			t.Fatal("Timed out waiting for deployment to be ready.")
+		}
+
+		deploymentResource, err := client.AppsV1().Deployments(namespaceName).Get(context.TODO(), "kubearchive", metav1.GetOptions{})
```

Comment:
I think that `context.Background()` should be used here instead
```suggestion
		deploymentResource, err := client.AppsV1().Deployments(namespaceName).Get(context.Background(), "kubearchive", metav1.GetOptions{})
```

from the [context docs](https://pkg.go.dev/context)

context.Background()

> It is typically used by the main function, initialization, and tests, and as the top-level Context for incoming requests

context.TODO()

> Code should use context.TODO when it's unclear which Context to use or it is not yet available (because the surrounding function has not yet been extended to accept a Context parameter).

Reply:
Done